grails-app/controllers/org/broadinstitute/orsp/SubmissionController.groovy

```diff
@@ -55,48 +48,61 @@ class SubmissionController extends AuthenticatedController {
             }
         }
         SubmissionType defaultType = SubmissionType.getForLabel(params.get("type").toString()) ?: SubmissionType.Amendment.label
-        render(view: '/submission/submission',
-                model: [issue:                      issue,
-                        submission:                 submission,
-                        docTypes:                   PROJECT_DOC_TYPES,
-                        submissionTypes:            submissionTypes,
-                        submissionNumberMaximums:   submissionNumberMaximums,
-                        defaultType:                defaultType
-                ])
+        render( [issue:                      issue,
+                 typeLabel:                  issue.typeLabel,
+                 submission:                 submission,
+                 documents:                  submission?.documents,
+                 docTypes:                   PROJECT_DOC_TYPES,
+                 submissionTypes:            submissionTypes,
+                 submissionNumberMaximums:   submissionNumberMaximums,
+                 defaultType:                defaultType
+        ] as JSON)
     }
 
     def save() {
-        Issue issue = queryService.findByKey(params.projectKey)
-        Submission submission
-        if (params?.submissionId) {
-            submission = Submission.findById(params?.submissionId)
-            submission.number = Integer.parseInt(params?.submissionNumber)
-            submission.type = params?.submissionType
-            submission.comments = params?.comments
-            if (!submission.save(flush: true)) {
-                flash.message = submission.errors.allErrors.collect { it }.join("<br/>")
+        JsonParser parser = new JsonParser()
+        JsonArray dataSubmission = parser.parse(request.parameterMap["submission"].toString())
+        List<MultipartFile> files = request.multiFileMap.collect { it.value }.flatten()
```

Comment:
I'm not sure about how you're using request here. This is an instance of `HttpServletRequest` which doesn't have a `multiFileMap` field/method. What this should be doing under the hood is using a `GrailsWebRequest` which knows if it's a `multipartRequest` request or not. If it is a multipartRequest, then Grails will store all of the elements of `multiFileMap` as elements of the parameterMap. I'm looking into a better way to do this at the moment, but the cast here is concerning. There might be an easy way to implement a trait that exposes this more readily to the controller. I looked at implementing `WebAttributes` but that wasn't very helpful, even though it gives you direct access to the `GrailsWebRequest`

Reply:
Should I try to implement a trait with something like this inside?
```MultiValueMap<String, Object> map = new LinkedMultiValueMap<>();
List<ByteArrayResource> files = new ArrayList<>();

for(MultipartFile file : multiPartFileList) {
files.add(new ByteArrayResource(file.getBytes()));
}
map.put("files", files);

Reply:
OK, there are two paths we can go down here. 

### MultiFileMap
In the master branch, we have a method to add a document to an issue that looks a lot like what you have in this PR. It's older code, but functional, so here is that version:
```
        try {
            if (request.multiFileMap?.files) {
                storageProviderService.saveMultipartFiles(
                        (String) getUser()?.displayName,
                        (String) getUser()?.userName,
                        (String) issue.projectKey,
                        (String) params.type,
                        (Collection<MultipartFile>) request.multiFileMap.files)
            }
            redirect(controller: 'irb', action: "show", params: [id: issue.projectKey, tab: 'documents'])
        } catch (Exception e) {
            render([status: 500, text: [error: e.message] as JSON])
        }
```

### DiskFileItem
In the original submission controller, we had a separate method for adding a file. This is newer code, and a slightly more preferable approach. What Grails does behind the scenes is taking a multipart request and mapping the file elements to the parameter map. That gives us `params.files` which may or may not exist:
```
            DiskFileItem file = params?.files?.part?.fileItem
            if (file) {
                StorageDocument document = storageProviderService.saveFileItem(
                        (String) getUser()?.displayName,
                        (String) getUser()?.userName,
                        (String) params?.projectKey,
                        (String) params?.type,
                        file
                )
                ....
             } else {
                flash.error = "Please select a file to upload"
            }
```